polaris-react/src/components/Badge/Badge.tsx

```diff
@@ -115,7 +127,7 @@ export function Badge({
     <VisuallyHidden>{accessibilityLabel}</VisuallyHidden>
   );
 
-  if (progressLabel) {
+  if (progressLabel && !icon) {
```

Comment:
Technically I don’t really need this change but I assume not all Polaris consumers use TypeScript, so at least it makes it prefer the icon to the progress if you pass both?